lib/Centreon/Object/Relation/Host/Template/Host.php

```diff
@@ -70,6 +70,45 @@ public function insert($fkey, $skey = null)
         $this->db->query($sql, array($fkey, $skey, $order));
     }
 
+    /**
+     * Delete host template / host relation and linked services
+     *
+     * @param int $fkey
+     * @param int $skey
+     * @return void
+     */
+    public function delete($fkey, $skey = null): void
+    {
+        $this->db->beginTransaction();
+        try {
+            parent::delete($fkey, $skey);
+            // Delete linked services :
+            // First get service IDs
+            $stmt = $this->db->prepare(
+                'SELECT service_service_id FROM host_service_relation WHERE host_host_id = :host_id'
+            );
+            $stmt->bindParam(':host_id', $fkey, PDO::PARAM_INT);
+            $stmt->execute();
+
+            foreach ($stmt->fetchAll(PDO::FETCH_ASSOC) as $row) {
+                $stmt2 = $this->db->prepare(
+                    'DELETE FROM host_service_relation
+                    WHERE host_host_id = :host_id
+                    AND service_service_id = :service_id'
+                );
+                             WHERE host_host_id = :host_id
+                             AND service_service_id = :service_id');
```

Comment:
instead of making a loop for each `service_service_id` occurrences, concatenate and bind the `service` in a "IN" clause.
eg : https://github.com/centreon/centreon-widget-service-monitoring/pull/125/files

Reply:
done.